tests/snappi_tests/multidut/pfc/files/m2o_oversubscribe_lossy_helper.py

```diff
@@ -0,0 +1,344 @@
+# uncompyle6 version 3.9.0
+# Python bytecode version base 2.7 (62211)
+# Decompiled from: Python 3.10.4 (tags/v3.10.4:9d38120, Mar 23 2022, 23:13:41) [MSC v.1929 64 bit (AMD64)]
+# Embedded file name: /var/johnar/sonic-mgmt/tests/snappi/multi_dut_rdma/files/rdma_helper.py
+# Compiled at: 2023-02-10 09:15:26
+from math import ceil                                                                               # noqa: F401
+import logging                                                                                      # noqa: F401
+from tests.common.helpers.assertions import pytest_assert, pytest_require                           # noqa: F401
+from tests.common.fixtures.conn_graph_facts import conn_graph_facts, fanout_graph_facts             # noqa: F401
+from tests.common.snappi_tests.snappi_helpers import get_dut_port_id                                # noqa: F401
+from tests.common.snappi_tests.common_helpers import pfc_class_enable_vector, \
+     stop_pfcwd, disable_packet_aging                                                               # noqa: F401
+from tests.common.snappi_tests.port import select_ports                                             # noqa: F401
+from tests.common.snappi_tests.snappi_test_params import SnappiTestParams
+from tests.common.snappi_tests.traffic_generation import setup_base_traffic_config, \
+     verify_m2o_oversubscribtion_results, run_traffic                                               # noqa: F401
+logger = logging.getLogger(__name__)
+
+PAUSE_FLOW_NAME = 'Pause Storm'
+TEST_FLOW_NAME = 'Test Flow'
+TEST_FLOW_AGGR_RATE_PERCENT = 30
+BG_FLOW_NAME = 'Background Flow'
+BG_FLOW_AGGR_RATE_PERCENT = 25
+DATA_PKT_SIZE = 1024
+DATA_FLOW_DURATION_SEC = 5
+DATA_FLOW_DELAY_SEC = 2
+SNAPPI_POLL_DELAY_SEC = 2
+TOLERANCE_THRESHOLD = 0.05
+
+
```

Comment:
please add comments about how the test is done and expected results.

Reply:
The background traffic will include two lossless traffic streams, with the SONiC default lossless priorities of 3 and 4, and each having a 25% bandwidth. The test data traffic will consist of two lossy traffic streams, each with randomly chosen priorities (0..2, 5..7), and randomized so that either both streams are at 30% bandwidth each, or for one stream to be at 25% while the other is at 30% bandwidth. Start the test traffic from IXIA Tx ports 1 & 2; the additional increase in bandwidth will be either 55% or 60% depending on randomization (one stream with 25% BW, second stream with 30%, or both at 30% each), for a total oversubscription of either 103% or 108%.
After a few seconds, measure the received bandwidth per flow on the IXIA Rx port 3: total egress port bandwidth should be 100%. Internally the router should be dropping lossy frames from the streams causing the over-subscription.